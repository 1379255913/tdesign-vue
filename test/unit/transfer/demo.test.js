import { mount } from '@vue/test-utils';
import path from 'path';

const rootDir = path.resolve(__dirname, '../../../');

function testDemo(demoName) {
  const demoFile = path.resolve(rootDir, 'examples/transfer/demos', `${demoName}.vue`);
  const demo = require(demoFile).default;
  it(`${demoName} demo works fine`, () => {
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
}

// unit test for component in examples.
describe('Transfer', () => {
  testDemo('base');
  testDemo('checked');
  testDemo('custom');
  testDemo('disabled');
  testDemo('empty');
  testDemo('pagination');
  testDemo('search');
  testDemo('target-value');
  testDemo('tree');
});
