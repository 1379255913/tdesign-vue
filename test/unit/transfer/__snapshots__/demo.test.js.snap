// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer base demo works fine 1`] = `
<div
  class="t-transfer"
>
  <div
    class="t-transfer-list t-transfer-list-source"
  >
    <div
      class="t-transfer-list__header"
    >
      <div>
        <label
          class="t-checkbox"
        >
          <input
            class="t-checkbox__former"
            type="checkbox"
            value=""
          />
          <span
            class="t-checkbox__input"
          />
          <span
            class="t-checkbox__label"
          />
        </label>
        <span>
          0 / 20项
        </span>
      </div>
      <span />
    </div>
    <div
      class="t-transfer-list__body"
    >
      <div
        class="t-transfer-list__content narrow-scrollbar"
      >
        <div
          class="t-checkbox-group"
        >
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled="disabled"
                name=""
                type="checkbox"
                value="0"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容1
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="1"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容2
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="2"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容3
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="3"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容4
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled="disabled"
                name=""
                type="checkbox"
                value="4"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容5
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="5"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容6
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="6"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容7
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="7"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容8
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled="disabled"
                name=""
                type="checkbox"
                value="8"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容9
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="9"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容10
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="10"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容11
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="11"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容12
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled="disabled"
                name=""
                type="checkbox"
                value="12"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容13
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="13"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容14
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="14"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容15
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="15"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容16
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox t-is-disabled"
            >
              <input
                class="t-checkbox__former"
                disabled="disabled"
                name=""
                type="checkbox"
                value="16"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容17
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="17"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容18
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="18"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容19
            </span>
          </li>
          <li
            class="t-transfer-list__item"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="19"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              />
            </label>
            <span>
              内容20
            </span>
          </li>
        </div>
      </div>
    </div>
    <div
      class="t-transfer-footer"
    />
  </div>
  <div
    class="t-transfer-operations"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-is-disabled t-button--icon-only"
      disabled="disabled"
      type="button"
    >
      <svg
        class="t-icon t-icon-chevron-right"
      >
        <use
          xlink:href="#t-icon-chevron-right"
        />
      </svg>
    </button>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-is-disabled t-button--icon-only"
      disabled="disabled"
      type="button"
    >
      <svg
        class="t-icon t-icon-chevron-left"
      >
        <use
          xlink:href="#t-icon-chevron-left"
        />
      </svg>
    </button>
  </div>
  <div
    class="t-transfer-list t-transfer-list-target"
  >
    <div
      class="t-transfer-list__header"
    >
      <div>
        <label
          class="t-checkbox"
        >
          <input
            class="t-checkbox__former"
            type="checkbox"
            value=""
          />
          <span
            class="t-checkbox__input"
          />
          <span
            class="t-checkbox__label"
          />
        </label>
        <span>
          0 / 0项
        </span>
      </div>
      <span />
    </div>
    <div
      class="t-transfer-list__body"
    >
      <div
        class="t-transfer-empty"
      >
        暂无数据
      </div>
    </div>
    <div
      class="t-transfer-footer"
    />
  </div>
</div>
`;
